base/promotion.jl

```diff
@@ -241,6 +252,20 @@ end
     ^(x, y)
 
 Exponentiation operator.
+
+```jldoctest
+julia> 3^5
+243
+
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> A^3
```

Comment:
this example's good, could use a corresponding word or two of description that it's the matrix power operator